pulp_container/app/registry.py

```diff
@@ -97,6 +97,8 @@ async def get_tag(self, request):
                 streamed back to the client.
 
         """
+        self._reset_db_connection()
```

Comment:
I was not sure where to reset the connection - directly in the plugin or core.
What would be the implications if we'd add `self._reset_db_connection()` instead in core directly in each method?
@daviddavis @mdellweg ?

Reply:
I think what you have here is correct (at least according to my original intent, which was that it should be called once at each entry point and not in helper functions such as `_match_distribution`). 

However, I was very leery about introducing this function originally and still not sure if it's a good idea. Having someone else who's familiar with the content app to review and weigh in would be good.

Reply:
I'd prefer to have that call being the first thing when a new request is handled. But i don't even see where in `Handler` that would be. But maybe it's safe and not a bit performance impact if we add it in more places.

Reply:
I see the core's content app has these two primary entry points to the code https://github.com/pulp/pulpcore/blob/master/pulpcore/content/__init__.py#L54-L55 and I see both of them already call the `self._reset_db_connection` so I think core's already got it. Agreed that I'm also a little unclear on why it's needed in the first place, but core has it.

I'm not very familiar with pulp_container's registry, but it seems like it adds these top-level new routes: https://github.com/pulp/pulp_container/blob/master/pulp_container/app/content.py#L8-L14

Reply:
pulp_container does not call into core's `stream_content` hence I needed to add `self._reset_db_connection` before `_math_distribution` is called.

Reply:
I think in general though, if pulpcore is already calling it in these handlers, then adding the calls here is unlikely to break anything.